xla/stream_executor/sycl/sycl_gpu_runtime.cc

```diff
@@ -107,4 +196,514 @@ absl::StatusOr<::sycl::device> SyclDevicePool::GetDevice(int device_ordinal) {
   return device_pool_[device_ordinal];
 }
 
+StreamPoolMap SyclStreamPool::stream_pool_map_;
+absl::Mutex SyclStreamPool::stream_pool_mu_(absl::kConstInit);
+
+static const ::sycl::async_handler SyclAsyncHandler =
+    [](::sycl::exception_list ex_list) {
+      for (auto& e : ex_list) {
+        try {
+          std::rethrow_exception(e);
+        } catch (::sycl::exception& e) {
+          LOG(ERROR) << "SYCL exception: " << e.what()
+                     << ", file = " << __FILE__ << ", line = " << __LINE__
+                     << ".";
+        }
+      }
+    };
```

Comment:
This define a global object of a non-trivial type (std::function). This is discouraged in the Google C++ style guide because it can lead to initialization and destruction order issues.

Since the lambda is stateless, it should be possible to just declare a function here:

```
void SyclAsyncHandler(::sycl::exception_list ex_list) {
// ...
}
```

The conversion into to a std::function will then happen when the function is getting passed to the sycl::queue constructor.